src/CryptoBank.WebAPI/Features/Deposits/Requests/GetDepositAddress.cs

```diff
@@ -0,0 +1,115 @@
+using CryptoBank.Database;
+using CryptoBank.Domain.Models;
+using CryptoBank.WebAPI.Common.Errors.Exceptions;
+using CryptoBank.WebAPI.Common.Services;
+using CryptoBank.WebAPI.Features.Deposits.Errors;
+using CryptoBank.WebAPI.Features.Deposits.Services;
+using JetBrains.Annotations;
+using MediatR;
+using Microsoft.EntityFrameworkCore;
+using NBitcoin;
+
+namespace CryptoBank.WebAPI.Features.Deposits.Requests;
+
+public class GetDepositAddress
+{
+    public record Request : IRequest<Response>;
+
+    public record Response(string CryptoAddress);
+
+    [UsedImplicitly]
+    public class RequestHandler : IRequestHandler<Request, Response>
+    {
+        private readonly CurrentAuthInfoSource _currentAuthInfoSource;
+        private readonly CryptoBankDbContext _dbContext;
+        private readonly NetworkSource _networkSource;
+
+        public RequestHandler(
+            CurrentAuthInfoSource currentAuthInfoSource,
+            CryptoBankDbContext dbContext,
+            NetworkSource networkSource)
+        {
+            _currentAuthInfoSource = currentAuthInfoSource;
+            _dbContext = dbContext;
+            _networkSource = networkSource;
+        }
+
+        public async Task<Response> Handle(Request request, CancellationToken cancellationToken)
+        {
+            var userId = _currentAuthInfoSource.GetUserId();
```

Comment:
It is strange for me. I think about request classes only as a representation of HttpRequest.

I understood the idea with cache, but shouldn't we split HttpRequest and MediatorRequest?

Reply:
I think we talk about the same ideas, but I didn't express my thoughts clearly. Because the most popular use case in my code is using `*Request` as an object for HTTP requests and MediatR requests. So, if I will add `UserId` to the request object, I create a false impression that `UserId` filling in an HTTP request. Because of it, I offered to create separate classes for HTTP requests and MediatR requests. But I have to do it for all request objects in the code.